### ISSUE TITLE
Support for Storybook 7 with Vite?

### ISSUE THREAD

**Issue by peterstriz** [SHO-ai-org/addon-stitches#1]
Hey, I really like your plugin and just want to ask, if there is a way to get this plugin working with the new version of Storybook? I tried to migrate my project, but found out that switching themes no longer works. The button for theme change is visible, but after pressing it, whole page freezes indefinitely.

Setup:
```
"@storybook/react": "^7.0.0-beta.50",
"@storybook/react-vite": "^7.0.0-beta.50",
"multiple-themes-stitches": "^0.0.10",
```

**Comment by mathisobadia** [comment]
@peterstriz Yes we plan on supporting it, it's not very clear yet how to upgrade the addon but we are looking into it. I hope to get something around next week.

**Comment by mathisobadia** [comment]
@peterstriz we just released version v1.0.0 that should work with storybook v7 beta / rc if you can test it. thanks !

**Comment by peterstriz** [comment]
Hey thanks for your quick work, but now it doesn't show the button for switching themes at all... 😢 

`preview.tsx`
```tsx
import React from 'react';
import { withDesign } from 'storybook-addon-designs';

import { Box } from '../components/box/box';
import { createGlobalStyles } from '../theme/createGlobalStyles';
import { theme, trustCenterTheme } from '../theme/stitches.config';
import { ThemesParameter } from 'multiple-themes-stitches';

const multipleThemesStitches: ThemesParameter = {
  values: [
    {
      name: 'Default',
      theme: theme,
    },
    {
      name: 'Trust-Center',
      theme: trustCenterTheme,
    },
  ],
};

export const decorators = [
  (Story) => {
    createGlobalStyles();

    return (
      <Box css={{ p: '$3' }}>
        <Story />
      </Box>
    );
  },
  withDesign,
];

export const parameters = {
  actions: { argTypesRegex: '^on[A-Z].*' },
  controls: {
    matchers: {
      // color: /(background|color)$/i,
      date: /Date$/,
    },
  },
  backgrounds: {
    default: 'dark',
    values: [
      {
        name: 'dark',
        value: theme.colors.grayMain.value,
      },
      {
        name: 'black',
        value: theme.colors.black100.value,
      },
      {
        name: 'white',
        value: theme.colors.white100.value,
      },
    ],
  },
  design: {
    type: 'figma',
  },
  chromatic: {
    // Allow CSS animations in Chromatic
    pauseAnimationAtEnd: true,
    viewports: [1800],
  },
  // set themes so we can manually change themes in stories
  multipleThemesStitches,
  options: {
    storySort: {
      order: ['Atoms', 'Molecules', 'Organisms', '*'],
    },
  },
};
```

`main.ts`
```ts
import type { StorybookConfig } from '@storybook/react-vite';
import turboSnap from 'vite-plugin-turbosnap';

const config: StorybookConfig = {
  stories: ['../components/**/*.mdx', '../components/**/*.stories.@(|ts|tsx)'],
  addons: [
    'multiple-themes-stitches',
    '@storybook/addon-essentials',
    '@storybook/addon-interactions',
    '@storybook/addon-links',
    'storybook-addon-designs',
  ],
  framework: {
    name: '@storybook/react-vite',
    options: {},
  },
  docs: {
    autodocs: 'tag',
  },
  async viteFinal(config, { configType }) {
    if (configType === 'PRODUCTION') {
      config.plugins?.push(turboSnap({ rootDir: config.root || '' }));
    }

    return config;
  },
};

export default config;
```

`package.json`

```json
{
  "dependencies": {
    "@fontsource/montserrat": "^4.5.12",
    "@fontsource/oswald": "^4.5.10",
    "@radix-ui/react-accordion": "^1.0.0",
    "@radix-ui/react-aspect-ratio": "^1.0.0",
    "@radix-ui/react-checkbox": "^1.0.0",
    "@radix-ui/react-collapsible": "^1.0.0",
    "@radix-ui/react-dialog": "^1.0.0",
    "@radix-ui/react-dropdown-menu": "^1.0.0",
    "@radix-ui/react-label": "^2.0.0",
    "@radix-ui/react-popover": "^1.0.0",
    "@radix-ui/react-primitive": "^1.0.0",
    "@radix-ui/react-radio-group": "^1.0.0",
    "@radix-ui/react-scroll-area": "^1.0.0",
    "@radix-ui/react-select": "^1.0.0",
    "@radix-ui/react-tabs": "^1.0.0",
    "@radix-ui/react-toast": "^1.0.0",
    "@radix-ui/react-tooltip": "^1.0.0",
    "@radix-ui/react-visually-hidden": "^1.0.0",
    "@storybook/testing-library": "^0.0.14-next.1",
    "@stitches/react": "1.3.1-1",
    "@tanstack/react-table": "^8.5.11",
    "multiple-themes-stitches": "^1.0.0",
    "buffer": "^6.0.3",
    "clsx": "^1.2.1",
    "csstype": "^3.1.0",
    "file-saver": "^2.0.5",
    "lodash-es": "^4.17.21",
    "magic-bytes.js": "^1.0.12",
    "mathjs": "^11.3.2",
    "mitt": "^3.0.0",
    "polished": "^4.2.2",
    "react-use": "^17.4.0",
    "swiper": "^8.3.2"
  },
  "devDependencies": {
    "@storybook/addon-actions": "7.0.0-rc.3",
    "@storybook/addon-essentials": "7.0.0-rc.3",
    "@storybook/addon-interactions": "7.0.0-rc.3",
    "@storybook/addon-links": "7.0.0-rc.3",
    "@storybook/addons": "7.0.0-rc.3",
    "@storybook/api": "7.0.0-rc.3",
    "@storybook/blocks": "7.0.0-rc.3",
    "@storybook/cli": "7.0.0-rc.3",
    "@storybook/components": "7.0.0-rc.3",
    "@storybook/core-events": "7.0.0-rc.3",
    "@storybook/jest": "^0.0.11-next.0",
    "@storybook/react": "7.0.0-rc.3",
    "@storybook/react-vite": "7.0.0-rc.3",
    "@storybook/theming": "7.0.0-rc.3",
    "@types/file-saver": "^2.0.5",
    "@types/lodash-es": "^4.17.6",
    "@types/node": "^17.0.34",
    "@types/react": "^ 18.0.21",
    "@types/react-dom": "^18.0.6",
    "@vitejs/plugin-react-swc": "^3.2.0",
    "chromatic": "^6.17.0",
    "jsdom": "^20.0.1",
    "npm-run-all": "^4.1.5",
    "prop-types": "^15.8.1",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "require-from-string": "^2.0.2",
    "rimraf": "^3.0.2",
    "storybook": "7.0.0-rc.3",
    "storybook-addon-designs": "^7.0.0-beta.2",
    "typescript": "^4.9.4",
    "vite": "^4.1.4",
    "vite-plugin-dts": "^1.7.3",
    "vite-plugin-svgr": "^2.4.0",
    "vite-plugin-turbosnap": "^1.0.1",
    "vitest": "^0.28.5"
  },
  "peerDependencies": {
    "react": "^18.0.0",
    "react-dom": "^18.0.0"
  }
}
```

**Comment by mathisobadia** [comment]
@peterstriz I think I fixed the issue can you test version 1.0.1 ?

**Comment by peterstriz** [comment]
It works!! Thanks a lot man! ❤️
